lib/internal/loader/ModuleRequest.js

```diff
@@ -36,24 +38,39 @@ loaders.set('esm', async (url) => {
 
 // Strategy for loading a node-style CommonJS module
 loaders.set('cjs', async (url) => {
-  return createDynamicModule(['default'], url, (reflect) => {
-    debug(`Loading CJSModule ${url}`);
-    const CJSModule = require('module');
-    const pathname = internalURLModule.getPathFromURL(new URL(url));
-    CJSModule._load(pathname);
+  debug(`Loading CJSModule ${url}`);
+  const CJSModule = require('module');
+  const pathname = internalURLModule.getPathFromURL(new URL(url));
+  const exports = CJSModule._load(pathname);
+  const es = exports[esModuleInterop] !== undefined ?
+    exports[esModuleInterop] : exports.__esModule;
+  const keys = es ? Object.keys(exports) : ['default'];
+  return createDynamicModule(keys, url, (reflect) => {
+    if (es) {
+      for (var i = 0; i < keys.length; i++)
+        reflect.exports[keys[i]].set(exports[keys[i]]);
+    } else {
+      reflect.exports.default.set(exports);
+    }
   });
 });
 
 // Strategy for loading a node builtin CommonJS module that isn't
 // through normal resolution
 loaders.set('builtin', async (url) => {
-  return createDynamicModule(['default'], url, (reflect) => {
-    debug(`Loading BuiltinModule ${url}`);
-    const exports = NativeModule.require(url.substr(5));
+  debug(`Loading BuiltinModule ${url}`);
+  const exports = NativeModule.require(url.substr(5));
+  const keys = Object.keys(exports);
+  return createDynamicModule(['default', ...keys], url, (reflect) => {
```

Comment:
It's not documented that builtin modules still have a default export.

Reply:
Perhaps for these native module keys we should add a filter - `Object.keys(exports).filter(name => name.startsWith('_') === false)`? This would avoid private keys such as `import { _makeLong } from 'path'` being on the public API that will end up in the list for type hinting systems such as TypeScript.

Reply:
Please do not; `_` may be used to convey hopeful privacy, but it’s always fully public.